__tests__/distributors/zulu-installer.test.ts

```diff
@@ -0,0 +1,139 @@
+import { HttpClient } from '@actions/http-client';
+import * as semver from 'semver';
+import { ZuluDistributor } from '../../src/distributors/zulu/installer';
+import { IZuluVersions } from '../../src/distributors/zulu/models';
+import * as utils from '../../src/util';
+
+const manifestData = require('../data/zulu-releases-default.json') as [];
+
+describe('getAvailableVersions', () => {
+  let spyHttpClient: jest.SpyInstance;
+  let spyUtilGetDownloadArchiveExtension: jest.SpyInstance;
+  let zuluDistributor: ZuluDistributor;
+
+  beforeEach(() => {
+    spyHttpClient = jest.spyOn(HttpClient.prototype, 'getJson');
+    spyHttpClient.mockReturnValue({
+      statusCode: 200,
+      headers: {},
+      result: manifestData as IZuluVersions[]
+    });
+
+    spyUtilGetDownloadArchiveExtension = jest.spyOn(utils, 'getDownloadArchiveExtension');
+    spyUtilGetDownloadArchiveExtension.mockReturnValue('tar.gz');
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it.each([
+    [
+      { version: '11', arch: 'x86', packageType: 'jdk' },
+      '?os=macos&ext=tar.gz&bundle_type=jdk&javafx=false&arch=x86&hw_bitness=32&release_status=ga'
+    ],
+    [
+      { version: '11-ea', arch: 'x86', packageType: 'jdk' },
+      '?os=macos&ext=tar.gz&bundle_type=jdk&javafx=false&arch=x86&hw_bitness=32&release_status=ea'
+    ],
+    [
+      { version: '8', arch: 'x64', packageType: 'jdk' },
+      '?os=macos&ext=tar.gz&bundle_type=jdk&javafx=false&arch=x86&hw_bitness=64&release_status=ga'
+    ],
+    [
+      { version: '8', arch: 'x64', packageType: 'jre' },
+      '?os=macos&ext=tar.gz&bundle_type=jre&javafx=false&arch=x86&hw_bitness=64&release_status=ga'
+    ],
+    [
+      { version: '8', arch: 'x64', packageType: 'jdk+fx' },
+      '?os=macos&ext=tar.gz&bundle_type=jdk&javafx=true&arch=x86&hw_bitness=64&release_status=ga&features=fx'
+    ],
+    [
+      { version: '8', arch: 'x64', packageType: 'jre+fx' },
+      '?os=macos&ext=tar.gz&bundle_type=jre&javafx=true&arch=x86&hw_bitness=64&release_status=ga&features=fx'
+    ]
+  ])('build correct url for %s -> $s', async (input, parsedUrl) => {
+    zuluDistributor = new ZuluDistributor(input);
+    zuluDistributor['getPlatformOption'] = () => 'macos';
+    const buildUrl = `https://api.azul.com/zulu/download/community/v1.0/bundles/${parsedUrl}`;
+    await zuluDistributor['getAvailableVersions']();
+    expect(spyHttpClient.mock.calls).toHaveLength(1);
+    expect(spyHttpClient.mock.calls[0][0]).toBe(buildUrl);
```

Comment:
please, split setup/action/expect in tests by new lines for easier reading

Reply:
Done. Thank you.